lib/cli/watch-run.js

```diff
@@ -54,29 +94,59 @@ module.exports = (mocha, {ui}, fileCollectParams) => {
     }
   };
 
-  const purge = () => {
-    watchFiles.forEach(Mocha.unloadFile);
-  };
-
-  loadAndRun();
-
-  const rerun = () => {
-    purge();
-    eraseLine();
-    mocha.suite = mocha.suite.clone();
-    mocha.suite.ctx = new Context();
-    mocha.ui(ui);
-    loadAndRun();
-  };
+  const scheduleRun = () => {
+    if (rerunScheduled) {
+      return;
+    }
 
-  utils.watch(watchFiles, () => {
-    runAgain = true;
+    rerunScheduled = true;
     if (runner) {
       runner.abort();
     } else {
       rerun();
     }
+  };
```

Comment:
If one of the watched files is changed while the tests are still running, the tests are aborted, but not rerun?
Is this the usual behavior of a watch functionality to abort, instead of waiting for the test end and then restarting?

Out of topic: I don't trust `runner.abort()`. I doubt that the hook pattern is correct and all `afterEach` / `after` hooks run for correct tear-down. This is prone to errors.

Reply:
This is the same behavior as before. I think it makes sense to abort the tests when a file changes. If you usually want to get quick feedback when your code changes.